src/gui/EntryPreviewWidget.cpp

```diff
@@ -240,14 +240,13 @@ void EntryPreviewWidget::updateEntryGeneralTab()
         m_ui->togglePasswordButton->setVisible(false);
     }
 
-    if (config()->get(Config::Security_HideNotes).toBool()) {
-        setEntryNotesVisible(false);
-        m_ui->toggleEntryNotesButton->setVisible(!m_ui->entryNotesTextEdit->toPlainText().isEmpty());
-        m_ui->toggleEntryNotesButton->setChecked(false);
-    } else {
-        setEntryNotesVisible(true);
-        m_ui->toggleEntryNotesButton->setVisible(false);
-    }
+    auto hasNotes = !m_currentEntry->notes().isEmpty();
+    auto hideNotes = config()->get(Config::Security_HideNotes).toBool();
+
+    m_ui->entryNotesTextEdit->setVisible(hasNotes);
+    setEntryNotesVisible(hasNotes && !hideNotes);
+    m_ui->toggleEntryNotesButton->setVisible(hasNotes && hideNotes && !m_ui->entryNotesTextEdit->toPlainText().isEmpty());
```

Comment:
Can ` && !m_ui->entryNotesTextEdit->toPlainText().isEmpty()` be part of `hasNotes` (line 243), or would that break line 246?

If you always initialize `entryNotesTextEdit` with the notes from the current entry, then it'll get reset to an empty string, and then only one of `entryNotesTextEdit->toPlainText().isEmpty()` or `m_currentEntry->notes().isEmpty()` should be enough (and you won't have to hide the text edit either).

Reply:
> Can && !m_ui->entryNotesTextEdit->toPlainText().isEmpty() be part of hasNotes (line 243), or would that break line 246?

It would break

> If you always initialize entryNotesTextEdit with the notes from the current entry, then it'll get reset to an empty string, and then only one of entryNotesTextEdit->toPlainText().isEmpty() or m_currentEntry->notes().isEmpty() should be enough (and you won't have to hide the text edit either).

That would be the ideal case, but I was trying to change the code as little as possible to fix this bug ASAP 🙃 . That would require changing the setEntryNotesVisible and the setNotesVisible (which is also called by group functionality) functions. There is some work that could be done to simplify things but I'm not sure it is in the scope of this PR.